### PR TITLE
2 0 media edit

### PR DESCRIPTION
**Pull request by ngilain**
[OO-FEATURE] Add media edit view
https://github.com/open-orchestra/open-orchestra/pull/1145
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/2097
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/344
https://github.com/open-orchestra/open-orchestra-media-bundle/pull/248
https://github.com/open-orchestra/open-orchestra-media-file-bundle/pull/31

### CODE REVIEW

File: MediaAdmin/MediaForm/Strategy/DefaultStrategy.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+     *
+     * @return bool
+     */
+    public function support(MediaInterface $media)
+    {
+        return true;
+    }
+
+    /**
+     * Get the form type associated tothe strategy $media
```
Comment (alavieille) [inline MediaAdmin/MediaForm/Strategy/DefaultStrategy.php]: to the strategy

File: MediaAdmin/MediaForm/Strategy/ImageStrategy.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return 'image_media_form_strategy';
+    }
+}
```
Comment (alavieille) [inline MediaAdmin/MediaForm/Strategy/ImageStrategy.php:122]: eof

File: MediaAdmin/Tests/MediaForm/Strategy/ImageStrategyTest.php [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+    /**
+     * test cropAlternative
+     *
+     * @param string $format
+     * @param array  $crop
+     *
+     * @dataProvider provideFormData
+     */
+    public function testCropAlternative(array $crop, $file, $cropCount, $overrideCount)
```
Comment (alavieille) [inline MediaAdmin/Tests/MediaForm/Strategy/ImageStrategyTest.php]: php doc

File: MediaAdmin/Tests/MediaForm/Strategy/ImageStrategyTest.php [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+    /**
+     * test overrideAlternative
+     *
+     * @param string $format
+     * @param array  $crop
+     *
+     * @dataProvider provideFormData
+     */
+    public function testOverrideAlternative(array $crop, $file, $cropCount, $overrideCount)
```
Comment (alavieille) [inline MediaAdmin/Tests/MediaForm/Strategy/ImageStrategyTest.php]: php doc

File: MediaAdmin/Tests/MediaForm/Strategy/ImageStrategyTest.php [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        Phake::verify($this->objectManager, Phake::times($cropCount + $overrideCount))->persist($media);
+        Phake::verify($this->objectManager, Phake::times($cropCount + $overrideCount))->flush();
+    }
+
+    /**
+     * @param string $value
+     *
+     * @return Phake_IMock
+     */
+    protected function addChildForm($parentForm, $childName, $childValue)
```
Comment (alavieille) [inline MediaAdmin/Tests/MediaForm/Strategy/ImageStrategyTest.php]: php doc

File: MediaAdminBundle/Form/Type/MediaImageType.php [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\Form\FormInterface;
+use OpenOrchestra\MediaFileBundle\Manager\MediaStorageManager;
+
+/**
+ * Class MediaImageType
+ */
+class MediaImageType extends MediaBaseType
```
Comment (alavieille) [inline MediaAdminBundle/Form/Type/MediaImageType.php:14]: extends MediaBaseType is redundant with get parent

File: MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/Router/Media/MediaRouter.js [hunk 7]
```diff
@@ -59,6 +67,25 @@ class MediaRouter extends OrchestraRouter
...
+
+    /**
+     * Edit media
+     */
+    editMedia(mediaType, mediaId) {
+        let url = Routing.generate('open_orchestra_media_admin_media_form', {mediaId: mediaId});
+        this._displayLoader(Application.getRegion('content'));
+        FormBuilder.createFormFromUrl(url, (form) => {
+            let formViewClass = MediaFormView;
+            if (typeof this._mediaViews[mediaType] !== 'undefined') {
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/Router/Media/MediaRouter.js]: isn't extensible.

File: MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaFormView.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+    /**
+     * @inheritdoc
+     */
+    render() {
+        let template = this._renderTemplate(this._template);
+        this.$el.html(template);
+        this._$formRegion = $('.form-edit', this.$el);
+        super.render();
+        this._onRenderEnd();
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediaFormView.js]: why no override render in child class parent and use super

**Comment by ngilain** [comment]
@alavieille : update done

File: MediaAdminBundle/Form/Type/MediaImageType.php (hunk 6 above)
Comment (ngilain) [reply MediaAdminBundle/Form/Type/MediaImageType.php:14]: @alavieille : I'll remove the parent::buildXXX

**Pull request merged**